grant-project/grant-monorepo
# Require message signature for auth

No associated issue.

### What This Does

Requires a web3 signature on signup / signin. This doesn't actually do much in the way of authentication, but demonstrates what the flow would be like for web3 sign ins. This doesn't work for the hardware wallets, and will require different handling once we need to support things that can't sign in browser (e.g. raw private keys.)

### Steps to Test

* Attempt to sign up, confirm you need to sign a message to do so
* Attempt to sign in, confirm you need to sign a message to do so
* Refresh page, confirm you were auth'd in without needing to sign a message

## Screenshots

<img width="481" alt="screen shot 2018-10-30 at 12 44 18 am" src="https://user-images.githubusercontent.com/649992/47696502-6f845900-dbdd-11e8-9449-5831ee320c5c.png">
